pkgs/top-level/perl-packages.nix

```
@@ -21324,16 +21324,16 @@ let
       url = "mirror://cpan/authors/id/S/SZ/SZBALINT/WWW-Curl-4.17.tar.gz";
       sha256 = "1fmp9aib1kaps9vhs4dwxn7b15kgnlz9f714bxvqsd1j1q8spzsj";
     };
-    patches = [ ../development/perl-modules/WWW-Curl-4.17-Skip-preprocessor-symbol-only-CURL_STRICTER.patch ];
+    patches = [
+      (fetchpatch {
+        url = "https://aur.archlinux.org/cgit/aur.git/plain/curl-7.71.0.patch?h=perl-www-curl&id=261d84887d736cc097abef61164339216fb79180";
+        sha256 = "1hiw5lkflfa93z5d6k8fnnml0r08c653bbvvb8zx6gcrlbrdalfs";
+        name = "WWWCurl-curl-7.71.0.patch";
+      })
+    ];
+    NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin "-Wno-return-type";
```

```suggestion
    CFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-return-type";
```
Regarding CFLAGS, see #79303
Regarding the warning, those are driven by compiler frontend, not the platform.

---

Thx :)

---

Hm, that didn't quite work.

---

I see rest of pkgs/top-level/perl-packages.nix uses NIX_CFLAGS_*. I thought perl respected CFLAGS, but I guess I was wrong.